tls/s2n_connection.c

```diff
@@ -557,6 +527,16 @@ int s2n_connection_get_client_hello_version(struct s2n_connection *conn)
     return conn->client_hello_version;
 }
 
+int s2n_connection_did_handshake_negotiate_client_auth(struct s2n_connection *conn)
+{
+    if ((conn->handshake.handshake_type & CLIENT_AUTH)
+        && !(conn->closed)
+        && (APPLICATION_DATA == s2n_conn_get_current_message_type(conn))) {
```

Comment:
How about macro/helper for "is_handshake_complete?" here? I bet future code will need to use it. Lots of code peeking at specific handshake message types would not be ideal if we ever want to modify the state machine.